### PR TITLE
Add release dialog

### PR DESCRIPTION
**Pull request by mouhsen-ibrahim**

### CODE REVIEW

File: services/frontend-service/src/ui/utils/store.tsx [hunk 1]
```diff
@@ -253,3 +258,6 @@ export const useCurrentlyDeployedAt = (application: string, version: number) =>
             )
             .map((e) => e.name)
     );
+
+// Get current release dialog
+export const useCurrentReleaseDialog = () => useReleaseDialog(({ app, version }) => [app, version]);
```
Comment (medo-freiheit) [inline services/frontend-service/src/ui/utils/store.tsx:263]: This is not quite true, you need to get the release data from the overview store, which contains commitMsg,locks,envs...etc

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        </ListItem>
+                        <Divider />
+                        <ListItem button>
+                            <ListItemText primary="env2" />
+                        </ListItem>
+                    </List>
+                </Dialog>
+            </div>
+        ) : (
+            <div></div>
```
Comment (medo-freiheit) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:63]:
```suggestion
            <div/>
```use single tag close if you not passing any children

File: services/frontend-service/src/ui/utils/store.tsx (hunk 1 above)
Comment (mouhsen-ibrahim) [reply services/frontend-service/src/ui/utils/store.tsx:263]: Where should I store the information about which app/version is displayed now or not? I added a second call to `useCurrentlyDeployedAt` to fetch the envs

**Review by leandro-freiheit — APPROVED** [review]

File: services/frontend-service/src/ui/Pages/Home/Home.tsx [hunk 3]
```diff
@@ -26,8 +33,18 @@ export const Home: React.FC = () => {
     const searchedApp = useSearchedApplications(filteredApps, appNameParam);
 
     const apps = Object.values(searchedApp);
+
+    const { app, version } = useReleaseDialog(({ app, version }) => ({ app, version }));
+
+    const envs = useCurrentlyDeployedAt(app, version);
+    let releaseInfo = useReleaseInfo(app, version);
```
Comment (medo-freiheit) [inline services/frontend-service/src/ui/Pages/Home/Home.tsx:40]:
```suggestion
    let releaseInfo = useReleaseInfo(app, version);
```
please don't use let, only const

**Review by medo-freiheit — CHANGES_REQUESTED** [review]

File: services/frontend-service/src/ui/Pages/Home/Home.tsx [hunk 4]
```diff
@@ -26,8 +33,18 @@ export const Home: React.FC = () => {
...
+
+    if (releaseInfo === undefined) {
```
Comment (medo-freiheit) [inline services/frontend-service/src/ui/Pages/Home/Home.tsx:42]: this should be handled from useReleaseInfo

File: services/frontend-service/src/ui/Pages/Home/Home.tsx [hunk 5]
```diff
@@ -26,8 +33,18 @@ export const Home: React.FC = () => {
...
+        releaseInfo = {};
+    }
     return (
         <main className="main-content">
+            <ReleaseDialog app={app} version={version} release={releaseInfo} envs={envs}></ReleaseDialog>
```
Comment (medo-freiheit) [inline services/frontend-service/src/ui/Pages/Home/Home.tsx:47]:
```suggestion
            <ReleaseDialog app={app} version={version} release={releaseInfo} envs={envs}/>
```
if there are no children, please use one closed tag only

File: services/frontend-service/src/ui/utils/store.tsx [hunk 6]
```diff
@@ -253,3 +258,12 @@ export const useCurrentlyDeployedAt = (application: string, version: number) =>
             )
             .map((e) => e.name)
     );
+
+// Get release information for a version
+export const useReleaseInfo = (app: string, version: number) =>
+    useOverview(({ applications }) => {
+        if (app !== '') {
```
Comment (medo-freiheit) [inline services/frontend-service/src/ui/utils/store.tsx:265]:
```suggestion
            return applications[app]?.releases.filter((release) => release.version === version)[0];
```

**Review by medo-freiheit — APPROVED** [review]

**Pull request merged**